adapters/dmx/dmx.go

```diff
@@ -327,3 +339,17 @@ func isDmxParams(t interface{}) bool {
 		return false
 	}
 }
+
+func getIdfa(request *openrtb.BidRequest) (string, bool) {
+	device := request.Device
+	if device.IFA != "" {
+		return device.IFA, true
+	}
+	return "empty", false
+}
```

Comment:
It's possible for `Device` to be `nil`. Please add a protection check. There's also no need to use the literal "empty" since it's discarded. Consider the following:
```
func getIDFA(request *openrtb.BidRequest) (string, bool) {
  if request.Device == nil {
    return nil, false
  }

  return request.Device.IFA, request.Device.IFA != ""
}
```